src/TraceEvent/Symbols/SymbolReader.cs

```diff
@@ -1201,6 +1247,97 @@ private long CopyStreamToFile(Stream fromStream, string fromUri, string fullDest
             return byteCount;
         }
 
+        /// <summary>
+        /// Get a file from a symbol server with MSFZ format support.
+        /// Checks both normal cache location and msfz0 subdirectory, downloads files using .new extension,
+        /// and properly places MSFZ files in msfz0 subdirectory.
+        /// </summary>
+        private string GetFileFromServerWithMsfzSupport(string urlForServer, string fileIndexPath, string targetPath)
+        {
```

Comment:
I was thinking that moving the msfz changes out of GetFileFromServer would be simple, but it is not.  Let's go ahead and merge them back into GetFileFromServer so that we don't have to have two paths.  Be sure to update the tests.